lms/djangoapps/courseware/courses.py

```diff
@@ -94,7 +94,7 @@ def get_course(course_id, depth=0):
     return course
 
 
-def get_course_with_access(user, action, course_key, depth=0, check_if_enrolled=False, check_survey_complete=True, check_if_authenticated=False):  # lint-amnesty, pylint: disable=line-too-long
+def get_course_with_access(user, action, course_key, depth=0, check_if_enrolled=False, check_survey_complete=True, check_if_authenticated=False, allow_not_started_courses=False):  # lint-amnesty, pylint: disable=line-too-long
```

Comment:
I realize you didn't introduce the long line or the pragma, but could you break this signature across multiple lines and remove the pragma.  Here and below as well.

https://docs.openedx.org/en/latest/developers/references/developer_guide/style_guides/python-guidelines.html#breaking-long-lines